tests/system/src/test/java/test/robot/javafx/stage/StageFocusTest.java

```diff
@@ -0,0 +1,141 @@
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class StageFocusTest extends VisualTestBase {
+
+    static CountDownLatch startupLatch;
+    static CountDownLatch eventReceivedLatch;
+
+    static final int STAGE_SIZE = 200;
+
+    static final int STAGE_X = 100;
+    static final int STAGE_Y = 100;
+
+    static final int TIMEOUT = 2000; // ms
+    static final double TOLERANCE = 0.07;
+
+    static final Color SCENE_COLOR = Color.LIGHTGREEN;
+
+    private Stage theStage = null;
+
+    @BeforeAll
+    public static void setupOnce() throws Exception {
+        startupLatch = new CountDownLatch(1);
+        eventReceivedLatch = new CountDownLatch(1);
+    }
+
+    /**
+     * Checks whether Stage is actually shown when calling show()
+     *
+     * Meant as a "canary" test of sorts to ensure other tests relying on
+     * Stage being actually shown and on foreground work fine.
+     */
+    @Test
+    public void testStageHasFocusAfterShow() throws InterruptedException {
+        Util.runAndWait(() -> {
+            theStage = getStage(false);
+
+            Group root = new Group();
+            Scene scene = new Scene(root, STAGE_SIZE, STAGE_SIZE);
+            scene.setFill(SCENE_COLOR);
+            scene.setOnKeyPressed(e -> {
+                if (e.getCode() == KeyCode.A) {
+                    eventReceivedLatch.countDown();
+                }
+            });
+
+            theStage.setScene(scene);
+            theStage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> {
+                Platform.runLater(() -> {
+                    theStage.setX(STAGE_X);
+                    theStage.setY(STAGE_Y);
+                    startupLatch.countDown();
+                });
+            });
+            theStage.show();
+        });
+
+        assertTrue(startupLatch.await(TIMEOUT, TimeUnit.MILLISECONDS), "Timeout waiting for test stage to be shown");
+
+        // check if isFocused returns true
+        assertTrue(
+            theStage.isFocused(),
+            "Stage.isFocused() returned false! Stage does not have focus after showing. Some tests might fail because of this. " +
+            "If that is the case, try re-running the tests with '--no-daemon' flag in Gradle."
+        );
+
+        // give UI a bit of time to finish showing transition
+        // ex. on Windows above latch is set despite the UI still "animating" the show
+        sleep(500);
+
+        // check if window is on top
+        Util.runAndWait(() -> {
+            WritableImage capture = getRobot().getScreenCapture(null, STAGE_X, STAGE_Y, STAGE_SIZE, STAGE_SIZE, false);
+            PixelReader captureReader = capture.getPixelReader();
+            for (int x = 0; x < STAGE_SIZE; ++x) {
+                for (int y = 0; y < STAGE_SIZE; ++y) {
```

Comment:
I was thinking of testing ever 5th or 10th pixel, but this code is fast enough.
If I quickly click on some window underneath the test window, the test correctly fails with 

org.opentest4j.AssertionFailedError: expected:rgba(144,238,144,255) but was:rgba(234,234,234,255)